miso-web/src/main/webapp/scripts/download/download.js

```diff
@@ -0,0 +1,172 @@
+        
+            if(payload.length > (1024*1024*1.999) && myBlob !== toString ){
+                payload=dataUrlToBlob(payload);
+                mimeType=payload.type || defaultMime;
+            }else{          
+                return navigator.msSaveBlob ?  // IE10 can't do a[download], only Blobs:
+                    navigator.msSaveBlob(dataUrlToBlob(payload), fileName) :
+                    saver(payload) ; // everyone else can save dataURLs un-processed
+            }
+            
+        }else{//not data url, is it a string with special needs?
+            if(/([\x80-\xff])/.test(payload)){            
+                var i=0, tempUiArr= new Uint8Array(payload.length), mx=tempUiArr.length;
+                for(i;i<mx;++i) tempUiArr[i]= payload.charCodeAt(i);
+                payload=new myBlob([tempUiArr], {type: mimeType});
+            }         
+        }
+        blob = payload instanceof myBlob ?
+            payload :
+            new myBlob([payload], {type: mimeType}) ;
+
+
+        function dataUrlToBlob(strUrl) {
+            var parts= strUrl.split(/[:;,]/),
+            type= parts[1],
+            indexDecoder = strUrl.indexOf("charset")>0 ? 3: 2,
+            decoder= parts[indexDecoder] == "base64" ? atob : decodeURIComponent,
+            binData= decoder( parts.pop() ),
+            mx= binData.length,
+            i= 0,
+            uiArr= new Uint8Array(mx);
+
+            for(i;i<mx;++i) uiArr[i]= binData.charCodeAt(i);
+
+            return new myBlob([uiArr], {type: type});
+         }
+
+        function saver(url, winMode){
+
+            if ('download' in anchor) { //html5 A[download]
+                anchor.href = url;
+                anchor.setAttribute("download", fileName);
+                anchor.className = "download-js-link";
+                anchor.innerHTML = "downloading...";
+                anchor.style.display = "none";
+                anchor.addEventListener('click', function(e) {
+                    e.stopPropagation();
+                    this.removeEventListener('click', arguments.callee);
+                });
+                document.body.appendChild(anchor);
+                setTimeout(function() {
+                    anchor.click();
+                    document.body.removeChild(anchor);
+                    if(winMode===true){setTimeout(function(){ self.URL.revokeObjectURL(anchor.href);}, 250 );}
+                }, 66);
+                return true;
+            }
+
+            // handle non-a[download] safari as best we can:
+            if(/(Version)\/(\d+)\.(\d+)(?:\.(\d+))?.*Safari\//.test(navigator.userAgent)) {
+                if(/^data:/.test(url))  url="data:"+url.replace(/^data:([\w\/\-\+]+)/, defaultMime);
+                if(!window.open(url)){ // popup blocked, offer direct download:
+                    if(confirm("Displaying New Document\n\nUse Save As... to download, then click back to return to this page.")){ location.href=url; }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this usage of confirm(...). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)